Benchmark/main.cpp

```diff
@@ -0,0 +1,174 @@
+#include "../SinglePointTransform_lib/Interpolator.h"
+#include <QDebug>
+#include <QGuiApplication>
+#include <algorithm>
+#include <chrono>
+#include <iostream>
+#include <vector>
+
+void myMessageOutput(QtMsgType type,
+                     [[maybe_unused]] const QMessageLogContext& context,
+                     const QString& msg)
+{
+    QByteArray localMsg = msg.toLocal8Bit();
+    QString typeStr = "OTHER";
+    switch (type)
+    {
+        case QtDebugMsg:
+            typeStr = "DBG";
+            break;
+        case QtInfoMsg:
+            typeStr = "INF";
+            break;
+        case QtWarningMsg:
+            typeStr = "WRN";
+            break;
+        case QtCriticalMsg:
+            typeStr = "CRT";
+            break;
+        case QtFatalMsg:
+            typeStr = "FTL";
+            break;
+    }
+    std::cout << QString("%1: %2")
+                     .arg(typeStr)
+                     .arg(localMsg.constData())
+                     .toStdString()
+              << std::endl;
+}
+
+void interpolatorMeasurement(int led_number, int led_size, int angle,
+                             const std::string& pixmap_path, QObject* q_obj)
+{
+    QPixmap pix_map{};
+    pix_map.load(QString::fromStdString(pixmap_path));
+    Interpolator interpolator_obj{q_obj};
```

Comment:
I suppose there is no need to feed parent property of Interpolator. It should not cause any impact on measurements.

Reply:
Now I understand: Interpolator class has no default parameter in constructor so you are forced to feed parent parameter of constructor.

Reply:
Or I don't undestand. :D There is default parameter but you are using `0` instead of `nullptr`.

Reply:
Solved.